imports.py

```diff
@@ -12,193 +10,283 @@
-			print(', '.join(sys_names))
-
-	def summarize_code_system_by_name(self, name: str):
-		"""Summarize a code system by name param"""
-		response: Dict = self.do_hapi_request("CodeSystem", {'name': name})
-		# if not decode_response_status_not_ok(response):
-		decode_code_system_json(response)
-
-	def summarize_code_system_by_system(self, system: str):
-		"""Summarize a code system by system param"""
-		response: Dict = self.do_hapi_request("CodeSystem", {'system': system})
-		# if not decode_response_status_not_ok(response):
-		decode_code_system_json(response)
-
-	def validate_code(self, system_id: str, code: str):
-		"""Validate that code is in code system"""
-		response: Dict = self.do_hapi_request(f'CodeSystem/{system_id}/$validate-code?code={code}')
-		valid = [x['valueBoolean'] for x in response['parameter'] if x['name'] == 'result'][0]
-		if valid:
-			label = [x['valueString'] for x in response['parameter'] if x['name'] == 'display'][0]
-			print(label)
-		else:
-			print('Invalid code')
-
-	def lookup_code(self, system: URI, code: str):
-		"""Lookup code within code system"""
-		response: Dict = self.do_hapi_request(f'CodeSystem/$lookup?system={system}&code={code}')
-		decode_lookup_response(response)
+class CacheFhir:
+    """ missing an invalidation mechanism, use with caution """
+    
+    @staticmethod
+    def encode_filename_str(x):
+        """ Convert such that can create friendly system filename.
+            The issue here is backwards from what you might expect.
+            
+        """
+        #delim = '*'
+        #return x.replace('/', f'{delim}slash')
+        return x.replace('/', '*slash')
+
+    @staticmethod
+    def decode_filename_str(x):
+        """Convert from friendly system filename"""
+        #return x.replace(f'{delim}slash', '/')
+        return x.replace('*slash', '/')
+
+    def url_and_params_to_filename(url: str, params: Dict) -> str:
+        """Convert to filename"""
+        x = url
+        if params:
+            x += '___params_'
+            for k, v in params.items():
+                x += '__' + k + '_' + v
+        filename = CacheFhir.encode_filename_str(x) + '.json'
+        return filename
+    
+    def cache_write(url: str, params: Dict, response: Dict):
+        """Write to cache"""
+        filename = CacheFhir.url_and_params_to_filename(url, params)
+        if not os.path.exists(CACHE_DIR):
+            os.makedirs(CACHE_DIR)
+        path = os.path.join(CACHE_DIR, filename)
+        with open(path, 'w') as f:
+            json.dump(response, f, indent=4)
+
+    def cache_read(url: str, params: Dict) -> Union[Dict, None]:
+        """Read from cache"""
+        filename = CacheFhir.url_and_params_to_filename(url, params)
+        path = os.path.join(CACHE_DIR, filename)
+        if os.path.exists(path):
+            with open(path, 'r') as f:
+                data: Dict = json.load(f)
+                return data
+        return None
+
 
 
+
```

Comment:
All the lines are here because I removed tabs.

Reply:
ah i see